lib/internal/perf/observe.js

```diff
@@ -448,28 +448,27 @@ function hasObserver(type) {
 
 
 function startPerf(target, key, context = {}) {
-  if (hasObserver(context.type)) {
-    target[key] = {
-      ...context,
-      startTime: process.hrtime(),
-    };
-  }
+  target[key] = {
```

Comment:
why this change?

Reply:
As @mscdex said. 
```
Removing the conditionals is going to unnecessarily make more work on the GC when there are no observers because of the (inline) object being created for every request.
```
So i move the `hasObserver` out of `startPerf` and `stopPerf` to avoid generating useless objects when these are no observer. And the caller need to call  `hasObserver` before call `startPerf` or `stopPerf`